signature/sign.go

```diff
@@ -73,9 +75,14 @@ func configureOptions(opts ...Option) options {
 	return options
 }
 
+type Key interface {
+	Algorithm() jwa.KeyAlgorithm
```

Comment:
I'm a bit confused, how could `crypto.Signer` implement this interface?

Reply:
A caller to `Sign` would supply a `key` having some concrete type that implements `Key`. Whether that value has a concrete type that _also_ implements `crypto.Signer` can be determined by the type switch.